### ISSUE TITLE
[WAITING ON PR MERGE] Desktop - Margin above Request Money chat report header

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#21503]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Open the Desktop App and Login
2. Navigate to any money request chat

## Expected Result:
The chat header should not contain a top margin

## Actual Result:
The chat header contains a top margin

## Workaround:
UNknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [x] MacOS / Desktop

**Version Number:** n/a
**Reproducible in staging?:**n/a
**Reproducible in production?:**  v1.3.31-3 (desktop)
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation
![Screenshot 2023-06-19 at 8 43 04 AM](https://github.com/Expensify/App/assets/93399543/81149223-26b0-48f8-8502-311ab4a0d84d)

**Expensify/Expensify Issue URL:**
**Issue reported by:** Sibtain Ali
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1687147774347499

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01b8eeb668f4618a9e</li>
        <li>Upwork Job ID: 1682845000617562112</li>
        <li>Last Price Increase: 2023-07-22</li>
    </ul>
</details>

**Comment by Pujan92** [comment]
### Please re-state the problem that we are trying to solve in this issue.
Header background color differs for IOU report in desktop.

### What is the root cause of that problem?
HeaderGap is applied to ScreenWrapper component and there we don't have any info about the report type. This issue exists for desktop and for moneyRequest report where header bg is not the same as chat report type.

### What changes do you think we should make in order to solve the problem?
We need to pass an extra prop called headerGapBackground for ScreenWrapper from ReportScreen here and pass the respective green color when the report type is iou.
https://github.com/Expensify/App/blob/6728589b1d0ff4defb428c260bbac2dbfdc042f5/src/pages/home/ReportScreen.js#L250
This prop we need to maintain and pass to `HeaderGap` component from `Screenwrapper` to apply dynamic bg color.
https://github.com/Expensify/App/blob/6728589b1d0ff4defb428c260bbac2dbfdc042f5/src/components/ScreenWrapper/index.js#L134

Ref - https://github.com/Expensify/App/pull/19433#issuecomment-1561307033

**Comment by jliexpensify** [comment]
I can't repro this on v1.3.32-5 even after logging out/in again:

![image](https://github.com/Expensify/App/assets/30815269/c09101bb-cde9-4580-90de-7530ba5fe8fa)

![image](https://github.com/Expensify/App/assets/30815269/9759b513-9729-472b-95c4-1b0bdbb5fad2)

@allroundexperts what version are you on?

It also seems odd that this is intentional, it's a bit jarring - but I'll wait for @shawnborton to weigh in before I close this GH.

**Comment by jliexpensify** [comment]
Oh, so my mistake: I didn't realise this was the Desktop version! I re-downloaded it (I've not been using Desktop) and can see it on v1.3.31-3:

![image](https://github.com/Expensify/App/assets/30815269/7f9b79e2-9091-4eed-9b1b-67e966cb16f5)

@shawnborton - before I apply the External label, can you confirm if this is intentional?

**Comment by shawnborton** [comment]
I agree that the top space is intentional, but it should match the color below it. That being said, we're redoing all of the headers for money requests/tasks/etc, so I think we could probably close this since changes are coming. 

Also, I could have sworn we reported this one like weeks ago as well? cc @JmillsExpensify @trjExpensify @grgia @thienlnam @jasperhuangg

**Comment by thienlnam** [comment]
Yes, this was implemented in the way as the task header so they have the same problem - though this problem shouldn't occur anymore once https://github.com/Expensify/App/issues/20844 we add this, so let's just wait for this to be done instead of trying to solve this

**Comment by jliexpensify** [comment]
Still on hold

**Comment by jliexpensify** [comment]
Still on HOLD

**Comment by thienlnam** [comment]
Linked PR was merged!

**Comment by jliexpensify** [comment]
Ah, looks like the issue is gone on Desktop!

![image](https://github.com/Expensify/App/assets/30815269/d7272752-b86f-4993-95d4-b5208a22a1dc)

**Comment by jliexpensify** [comment]
@allroundexperts since you qualify for a reporting bonus, do you want to be paid via New.Dot?

**Comment by Pujan92** [comment]
@jliexpensify the issue is for money request report which still exists due to diff background color.

<img width="1201" alt="Screenshot 2023-07-14 at 12 07 44 AM" src="https://github.com/Expensify/App/assets/14358475/5657af6a-c3eb-4154-ba50-9519242324a3">

**Comment by jliexpensify** [comment]
@Pujan92 looks like you're on Dev, and I'm on Production Desktop. @thienlnam do you mind having a look at this issue?

**Comment by grgia** [comment]
This will be fixed in https://github.com/Expensify/App/pull/22484 / https://github.com/Expensify/App/pull/22562

**Comment by jliexpensify** [comment]
Nice, thanks @grgia !

**Comment by jliexpensify** [comment]
Looks like [these](https://github.com/Expensify/App/issues/21503#issuecomment-1635734942) were deployed to prod over the weekend, will re-test now.

**Comment by jliexpensify** [comment]
I'm on v1.3.43-7 for the Mac Desktop app and can't see it anymore:

![image](https://github.com/Expensify/App/assets/30815269/c37c34e2-87cf-45a1-b7d4-bc1fa13da7c6)
